### ISSUE TITLE
Issue: Unable to Login to X.com API Using twikit Client - LoginFlow Not Accessible (Error 400)

### ISSUE THREAD

**Issue by aqbspl** [d60/twikit#227]
I am trying to authenticate to X.com (formerly Twitter) using the twikit Python client. While attempting to log in using my credentials or cookies, I consistently encounter the following error:
```status: 400, message: "{"errors":[{"code":366,"message":"flow name LoginFlow is currently not accessible"}]}"```
**Steps to Reproduce**
Load or attempt to log in with credentials or cookies.
Run the script using twikit's Client.login() function.
**Example Code Snippet**
```
from twikit import Client
import asyncio
from configparser import ConfigParser

# Load credentials
config = ConfigParser()
config.read('config.ini')
username = config['X']['username']
email = config['X']['email']
password = config['X']['password']

client = Client(language='en-US')

async def main():
    try:
        await client.login(auth_info_1=username, auth_info_2=email, password=password)
        client.save_cookies('cookies.json')
        print("Login successful, cookies saved.")
    except Exception as e:
        print(f"Login failed: {e}")

# Run the async function
asyncio.run(main())
```
**Error Message**
```status: 400, message: "{"errors":[{"code":366,"message":"flow name LoginFlow is currently not accessible"}]}"```
**Additional Information**

- **Environment:** Ubuntu 22 LTS, Python 3.10
- **Library version:** twikit vX.X.X (latest version)
- **X.com API Access:** Standard user account (no developer API access)
**Tried solutions:**

- Using valid credentials

**Expected Behavior**
Successful login and subsequent authentication using either credentials or cookies.

**Actual Behavior**
The login process fails with a 400 error, and the LoginFlow is reported as "not accessible".

**Request for Help**

- Is this a known issue related to changes in X.com's authentication flow?
- Any workarounds or solutions for this issue would be greatly appreciated.

**Comment by PhatStraw** [comment]
if os.path.exists("cookies.json"):
        client.load_cookies("cookies.json")
    else:
        await client.login(
            auth_info_1=AUTH_INFO_1, auth_info_2=AUTH_INFO_2, password=PASSWORD
        )
        client.save_cookies("cookies.json")
        
        load the cookie

**Comment by aqbspl** [comment]
> ```
> if os.path.exists("cookies.json"):
>         client.load_cookies("cookies.json")
>     else:
>         await client.login(
>             auth_info_1=AUTH_INFO_1, auth_info_2=AUTH_INFO_2, password=PASSWORD
>         )
>         client.save_cookies("cookies.json")
> ```
> 
> ```
>     load the cookie
> ```

Thanks for the suggestion!

The issue I'm facing occurs during the login process itself. When I try to log in using credentials, I receive the following error:
`status: 400, message: "{"errors":[{"code":366,"message":"flow name LoginFlow is currently not accessible"}]}"
`

Since the login isn't successful, the `client.save_cookies()` step doesn't execute, so I can't generate or use the `cookies.json` file. That’s why even if I use a check for the existence of `cookies.json`, it doesn't resolve the underlying issue.

Do you have any suggestions on how to handle this or bypass the login issue?

**Comment by aqbspl** [comment]
> did you solve the issue? @aqbspl

Not yet, @mohanad-hafez I am getting the same error

**Comment by jwaitzel** [comment]
Same issue here

**Comment by mohanad-hafez** [comment]
was anyone able to solve this? or did you find other alternatives?

**Comment by zecarreira** [comment]
I'm trying it today, and I have the same issue.. any idea?

**Comment by Strookyy** [comment]
use this repo:
https://github.com/mdmrcglu/twikit

worked for me, just install with:
pip3 install --upgrade git+https://github.com/mdmrcglu/twikit.git  

theres a pull request to merge this into the main branch

**Comment by dvilelaf** [comment]
> use this repo: https://github.com/mdmrcglu/twikit
> 
> worked for me, just install with: pip3 install --upgrade git+https://github.com/mdmrcglu/twikit.git
> 
> theres a pull request to merge this into the main branch

This solution produces the following exception for me on `utils.py:103`:

```
    return self.response['subtasks'][0]['subtask_id']
IndexError: list index out of range
```

**Comment by IK-O** [comment]
I installed with: pip3 install --upgrade git+https://github.com/mdmrcglu/twikit.git

but same error occurs.

    return self.response['subtasks'][0]['subtask_id']
    IndexError: list index out of range

**Comment by szsyzx** [comment]
Is there no solution to this problem？

**Comment by eatpwsyy** [comment]
> use this repo: https://github.com/mdmrcglu/twikit
> 
> worked for me, just install with: pip3 install --upgrade git+https://github.com/mdmrcglu/twikit.git
> 
> theres a pull request to merge this into the main branch

I use this but

**Comment by desktop69** [comment]
> > use this repo: https://github.com/mdmrcglu/twikit
> > worked for me, just install with: pip3 install --upgrade git+https://github.com/mdmrcglu/twikit.git
> > theres a pull request to merge this into the main branch
> 
> This solution produces the following exception for me on `utils.py:103`:
> 
> ```
>     return self.response['subtasks'][0]['subtask_id']
> IndexError: list index out of range
> ```

hey @dvilelaf  did you solve that problem ?

**Comment by SheebSuka** [comment]
Perhaps this solution will help you figure out the problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by dvilelaf** [comment]
Not really, both repos show issues for me, but not always. Sometimes it just works. For now, i'm logging via cookies. Looking forward for the fix PRs to be merged.

**Comment by brownepres** [comment]
How do you log via cookies? Is there a tutorial somewhere which I missed? @dvilelaf

**Comment by dvilelaf** [comment]
> How do you log via cookies? Is there a tutorial somewhere which I missed? @dvilelaf

https://github.com/d60/twikit/issues/227#issuecomment-2394925822

You will need to extract the cookies from your browser

**Comment by d60** [comment]
fixed in v2.2.0
